zerver/lib/import_realm.py

```diff
@@ -247,11 +247,20 @@ def fix_upload_links(data: TableData, message_table: TableName) -> None:
                     )
 
 
-def fix_stream_permission_group_settings(data: TableData, realm: Realm) -> None:
+def fix_stream_permission_group_settings(
+    data: TableData, role_system_groups_dict: dict[int, NamedUserGroup]
+) -> None:
+    system_groups_name_dict: dict[str, NamedUserGroup] = dict()
+    for role in NamedUserGroup.SYSTEM_USER_GROUP_ROLE_MAP:
+        group_name = NamedUserGroup.SYSTEM_USER_GROUP_ROLE_MAP[role]["name"]
+        system_groups_name_dict[group_name] = role_system_groups_dict[role]
```

Comment:
We already have `system_groups_name_dict` in `create_system_user_groups_for_realm` which is then used to calculate `role_system_groups_dict`.

Would it be better for `create_system_user_groups_for_realm` to return `system_groups_name_dict` and instead have `add_users_to_system_user_groups` compute `role_system_groups_dict`?

Reply:
Fixed!